Sources/UserDefaultsDependency/UserDefaultsDependency.swift

```diff
@@ -272,12 +272,12 @@ extension UserDefaults.Dependency: TestDependencyKey {
     return UserDefaults.Dependency { key, _ in
       storage.value[key]
     } set: { value, key in
-      var valueDidChange = false
+      let valueDidChange = LockIsolated(false)
       storage.withValue {
-        valueDidChange = !_isEqual($0[key], value)
+        valueDidChange.setValue(!_isEqual($0[key], value))
```

Comment:
FYI you could probably mitigate the need for `LockIsolated` by returning `!_isEqual(…)` from the closure instead of capturing a mutable var